lib/protobuf/decoder.ex

```diff
@@ -133,8 +133,24 @@ defmodule Protobuf.Decoder do
   end
 
   defdecoderp decode_delimited(field_number, message, props) do
-    <<bytes::bytes-size(value), rest::bits>> = rest
-    handle_value(rest, field_number, wire_delimited(), bytes, message, props)
+    bytes_remaining = byte_size(rest)
+
+    if value <= bytes_remaining do
```

Comment:
**nitpick**: for a little less churn, could we reverse the check and leave the happy path untouched at the end?

```elixir
bytes_remaining = byte_size(rest)

if value > bytes_remaining do
  # …
  raise Protobuf.DecodeError
end

<<bytes::bytes-size(value), rest::bits>> = rest
handle_value(rest, field_number, wire_delimited(), bytes, message, props)
```